pkg/specgen/generate/kube/kube.go

```diff
@@ -126,13 +125,31 @@ func ToSpecGen(ctx context.Context, containerYAML v1.Container, iid string, newI
 			s.StopSignal = &stopSignal
 		}
 	}
-	if len(containerYAML.Command) != 0 {
-		s.Command = containerYAML.Command
-	}
+
 	// doc https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#notes
-	if len(containerYAML.Args) != 0 {
+
+	if len(containerYAML.Command) == 0 && len(containerYAML.Args) == 0 {
```

Comment:
btw, I am pretty sure that the initialization with `[]string{}` from L109 is not necessary. It runs fine here without and `s.ContainerBasicConfig.Command` and `.Entrypoint`, which in theory should suffer from the same problem, are initialized just fine.